here-naksha-lib-jbon/src/jvmTest/kotlin/com/here/naksha/lib/jbon/JsonConverter.kt

```diff
@@ -0,0 +1,57 @@
+package com.here.naksha.lib.jbon
+
+import java.lang.UnsupportedOperationException
+
+object JsonConverter {
+
+    fun jsonToJbonByteArray(json: String): ByteArray {
+        val jsonObject = JbSession.get().parse(json)
+        val builder = JbSession.get().newBuilder()
+        when (jsonObject) {
+            is Map<*, *> -> writeMap(jsonObject, builder)
+            is Collection<*> -> writeCollection(jsonObject, builder)
+            else -> throw RuntimeException("not a json")
+        }
+        return builder.buildFeature(null)
+    }
+
+
+    fun writeMap(jsonMap: Map<*, *>, jbBuilder: JbBuilder) {
+        val start = jbBuilder.startMap()
+        for (entry in jsonMap.entries) {
+            jbBuilder.writeKey(entry.key as String)
+            writeValue(entry.value, jbBuilder)
+        }
+        jbBuilder.endMap(start)
+    }
+
+    fun writeCollection(collection: Collection<*>, jbBuilder: JbBuilder) {
+        val start = jbBuilder.startArray()
+        for (entry in collection) {
+            writeValue(entry, jbBuilder)
+        }
+        jbBuilder.endArray(start)
+    }
+
+    private fun writeValue(value: Any?, jbBuilder: JbBuilder) {
+        when (value) {
+            is String -> jbBuilder.writeString(value)
+            is Map<*, *> -> writeMap(value, jbBuilder)
+            is Boolean -> jbBuilder.writeBool(value)
+            is Int -> jbBuilder.writeInt32(value)
+            is Long -> jbBuilder.writeInt64(value)
+            is Double ->
+                if (value <= Float.MAX_VALUE && value >= Float.MIN_VALUE) {
+                    jbBuilder.writeFloat32(value.toFloat())
+                } else {
+                    jbBuilder.writeFloat64(value)
+                }
```

Comment:
this might be wrong somehow.
Common json parsers like Jackson always read floating point types as double no matter how big or small it is.
This code writes to bytea value using smallest posible type (float vs double), but it means that different features in table, under same path might have different type float or double.
Maybe it's not an issue because getDouble() can read both float and double, but maybe there is other scenario where it would fail.

The other approach would be to use only doubles, or to use dictionary as type provider